### PR TITLE
feat: support `LargeList` in `array_element`

### PR DESCRIPTION
**Pull request by Weijun-H**
## Which issue does this PR close?

Parts #8185

### CODE REVIEW

File: datafusion/physical-expr/src/array_expressions.rs [hunk 1]
```diff
@@ -369,56 +369,62 @@ pub fn make_array(arrays: &[ArrayRef]) -> Result<ArrayRef> {
...
     let mut mutable =
         MutableArrayData::with_capacities(vec![&original_data], true, capacity);
 
-    fn adjusted_array_index(index: i64, len: usize) -> Option<i64> {
+    fn adjusted_array_index<O: OffsetSizeTrait>(index: i64, len: O) -> Result<Option<O>>
+    where
+        i64: TryInto<O>,
+    {
+        let index: O = index.try_into().map_err(|_| {
+            DataFusionError::Execution(format!(
```
Comment (jayzhan211) [inline datafusion/physical-expr/src/array_expressions.rs:392]: nit: macro

**Review by jayzhan211 — APPROVED** [review]
LGTM

File: datafusion/physical-expr/src/array_expressions.rs (hunk 1 above)
Comment (Weijun-H) [reply datafusion/physical-expr/src/array_expressions.rs:392]: We cannot use marco here because of `map_err`

**Review by alamb — APPROVED** [review]
Thank you @Weijun-H  and @jayzhan211

**Pull request merged**